samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/core/subscriptions/EventStoreDBSubscriptionToAll.java

```diff
@@ -0,0 +1,181 @@
+        .fromPosition(new Position(checkpoint.get(), checkpoint.get()));
+    }
+    else {
+      this.subscriptionOptions.subscribeToAllOptions()
+        .fromStart();
+    }
+
+    System.out.println("Subscription to all '%s'".formatted(subscriptionOptions.subscriptionId()));
+
+    subscription = eventStoreClient.subscribeToAll(
+      listener,
+      this.subscriptionOptions.subscribeToAllOptions()
+    ).get();
+
+    isRunning = true;
+  }
+
+  public void stop(){
+    if(!isRunning)
+      return;
+
+    isRunning = false;
+    subscription.stop();
+  }
+
+  public boolean isRunning() {
+    return this.isRunning;
+  }
+
+  private void handleEvent(Subscription subscription, ResolvedEvent resolvedEvent) throws ExecutionException, InterruptedException {
+    try {
+      if (isEventWithEmptyData(resolvedEvent) || isCheckpointEvent(resolvedEvent))
+        return;
+
+      var eventClass = EventTypeMapper.toClass(resolvedEvent.getEvent().getEventType());
+
+      var streamEvent = EventEnvelope.of(eventClass, resolvedEvent);
+
+      if (streamEvent == null) {
+        // That can happen if we're sharing database between modules.
+        // If we're subscribing to all and not filtering out events from other modules,
+        // then we might get events that are from other module and we might not be able to deserialize them.
+        // In that case it's safe to ignore deserialization error.
+        // You may add more sophisticated logic checking if it should be ignored or not.
+        System.out.println("Couldn't deserialize event with id: %s".formatted(resolvedEvent.getEvent().getEventId()));
+
+        if (!subscriptionOptions.ignoreDeserializationErrors())
+          throw new IllegalStateException(
+            "Unable to deserialize event %s with id: %s"
+              .formatted(resolvedEvent.getEvent().getEventType(), resolvedEvent.getEvent().getEventId())
+          );
+
+        return;
+      }
+
+      // publish event to internal event bus
+      eventBus.publish(eventClass, streamEvent);
+
+      checkpointRepository.store(
+        this.subscriptionOptions.subscriptionId(),
+        resolvedEvent.getEvent().getPosition().getCommitUnsigned()
+      );
+    } catch (Exception e) {
+      System.out.println("Error consuming message: {%s}".formatted(e.getMessage()));
+      e.printStackTrace();
+      // if you're fine with dropping some events instead of stopping subscription
+      // then you can add some logic if error should be ignored
+      throw e;
+    }
+  }
+
+  private void handleDrop(Subscription subscription, Throwable throwable) {
+    System.out.println("Subscription was dropped due to " + throwable.getMessage());
+
+    resubscribe();
+  }
+
+  private void resubscribe() {
+    // You may consider adding a max resubscribe count if you want to fail process
+    // instead of retrying until database is up
+    while (true) {
+      var resubscribed = false;
+      synchronized (resubscribeLock) {
```

Comment:
There is some place to polish. Maybe with some Thread Pool / Thread Executor or someting like that :)

Reply:
This part is to make sure that there won't be more than one try at a time to resubscribe certain subscriptions. It's more around pessimistic locking than multithreading. Do you think that Thread Pool or Thread Executor will help me with that?

Reply:
I throw away my custom, hand-made resubscribe logic and used [Retry](https://docs.spring.io/spring-batch/docs/current/reference/html/retry.html) from Spring Boot 😃

Reply:
Even better! :) I had used Retry in one project and forgot that it exists.